frontend/src/scenes/billing/billingLogic.ts

```diff
@@ -256,8 +256,13 @@ export const billingLogic = kea<billingLogicType>([
                         )}% of your ${
                             productApproachingLimit.usage_key && productApproachingLimit.usage_key.toLowerCase()
                         } allocation.`,
+                        dismissKey: 'usage-limit-approaching',
                     }
                 }
+
+                lemonBannerLogic({ dismissKey: 'usage-limit-approaching' }).mount()
+                lemonBannerLogic({ dismissKey: 'usage-limit-approaching' }).actions.resetDismissKey()
+                lemonBannerLogic({ dismissKey: 'usage-limit-approaching' }).unmount()
```

Comment:
Can you ask in dev if there is a better way to do this? It feels like there should be. If not then ✅

Reply:
No response in slack so think it's 🆗

Reply:
Heya, sorry a bit late, there's usually no good reason to be mounting logics yourself, let kea handle it for you. Here, you'd want to `connect` this lemonBannerLogic (and the one above) -> https://keajs.org/docs/meta/connect/#pull-in-actions-and-values-from-keyed-logic and then just call actions.resetDismissKey(). It does not matter whether the banner exists yet or no, if it will have the same key, they'll go to the same logic.

A bigger problem here is that it appears you're calling an action inside a selector. This will be buggy and often not do the right thing because selectors are cached and not re-computed until inputs change, and are intended to be pure functions operating on the state, without changing the state.

All actions in a logic ought to happen via a listener. In this specific case, it seems like you want the alert to reset when the dismiss type changes. Probably the easiest way to do this is via a subscription: You subscribe to changes to the billingAlert value, and call the reset action based on the value.

Similar restrictions apply to selectors as reducers, see here: https://keajs.org/docs/core/reducers

Reply:
I spent some time scratching my head as to why a subscription implementation wasn't properly getting current and previous values across refreshes only to realize that the state of the selector is obviously reset on refresh. 

The only times in which billing is reloaded are: afterMount , billing limit updated, or product deactivated. I don't see a reason to change that behavior, so decided to create a reducer for billingAlert and update that reducer through a listener on loadBillingSuccess. This still doesn't work. The lemonBannerLogic, despite my having connected it to billingLogic, is never mounted. See: https://github.com/PostHog/posthog/pull/20675/files